scripts/extensions/videoEditor/src/VideoEditorThumbnail.tsx

```diff
@@ -0,0 +1,361 @@
+import * as React from 'react';
+import {ISuperdesk, IArticle} from 'superdesk-api';
+import VideoEditorContext from './VideoEditorContext';
+import {IVideoEditor} from './interfaces';
+import {pick} from 'lodash';
+
+interface IProps {
+    videoRef: React.RefObject<HTMLVideoElement>;
+    article: IArticle;
+    crop: IVideoEditor['crop'];
+    rotate: IVideoEditor['degree'];
+    onToggleLoading: (isLoading: boolean, loadingText?: string) => void;
+    onSave: (article: IArticle) => void;
+    onError: (err: any) => void;
+    getCropRotate: (crop: IVideoEditor['crop']) => IVideoEditor['crop'];
+}
+
+interface IState {
+    dirty: boolean;
+    type: 'capture' | 'upload' | '';
+    value: number | File; // capture positon or uploaded File
+    // most recent updated thumbnail, for re-drawing thumbnail when reset changes
+    thumbnail: string;
```

Comment:
We update article on saving thumbnail so it's redundant now, will remove it.
Previously, if video is just uploaded, article thumbnail is null even after updating thumbnail (because we think article does not need to be updated, but it is, because there is etag comparison)